src/hotspot/src/os/linux/vm/os_linux.cpp

```diff
@@ -581,6 +580,13 @@ void os::Linux::hotspot_sigmask(Thread* thread) {
 // detecting pthread library
 
 void os::Linux::libpthread_init() {
+#if 1
```

Comment:
Is that possible to use #ifdef a macro which is defined in musl only?

you comment out glibc inspection here. I think it would affect generic linux building. 
another approach is making this function support non-glibc.